### DISCUSSION TITLE
Licence Question

### DISCUSSION THREAD

**Discussion by Momo1987-cmd** [pymupdf/PyMuPDF#971] [Q&A] [answered]
Hi,
I have a question about the AGPL. I use PyMuPDF in my software to convert PDFs to PNGs. Does my software have to be open source? Or do I only have to do this if I change something in the library?

**Comment by JorjMcKie** [comment, marked as answer]
To comply with the open source AGPL you must remain open source and freeware.
If your software uses PyMuPDF (and hence also MuPDF) and you market it commercially, you do no longer fall under the AGPL.
To be sure please contact [Artifex](https://artifex.com/contact/). They will find an attractive solution I am sure.

**Reply by JorjMcKie** [reply]
If you offer your solution for free, it suffices to communicate the access to the source code and state that you also are subject to the AGPL.
But I am not a lawyer and cannot give legal advice.

**Reply by Momo1987-cmd** [reply]
Thanks for your answer. What would it look like if I offered the software free and open source, but had an update function in it. This update function would only be available with a key that you would have to buy. It would not update the program, but only a certain profile file. Is that in line with the AGPL?

**Reply by JorjMcKie** [reply]
No it is not. It would presumably have an impact on how Artifex would calculate any license fee, though.
Agin, I can only recommend talking to them about all those details - you should find there is much flexibility working with them.

**Reply by guillaume-chervet** [reply]
Hi @JorjMcKie ,

I just found PyMuPDF usage in langchain project where license is MIT.
https://github.com/langchain-ai/langchain/blob/master/libs/langchain/langchain/document_loaders/pdf.py

- Do you know if Lanchain can remain in MIT License ?

- If I use Langchain which use PyMuPDF for internal entreprise application (not commercial). Does software have to be open source ?

**Reply by JorjMcKie** [reply]
Anyone / anything using PyMuPDF must be either licensed as GNU AGPL v3.0 or commercial.
If AGPL then the app must be freeware and open-source - as per the license text. This should be no problem with internal uses, should it?

**Reply by guillaume-chervet** [reply]
Thank you @JorjMcKie for your information. I have requested a contact on Artifex website.

Should Lanchain need to switch to  GNU AGPL v3.0 license instead of MIT ?

**Reply by JorjMcKie** [reply]
In the end, Artifex Sales will need to answer this.
As I am no lawyer, you can only have my private optinion. And for me the license text amounts to a pretty clear "yes".
